src/main/java/org/jetbrains/research/refactorinsight/folding/RefactoringFolder.java

```diff
@@ -0,0 +1,166 @@
+package org.jetbrains.research.refactorinsight.folding;
+
+import com.intellij.application.options.colors.highlighting.RendererWrapper;
+import com.intellij.codeInsight.daemon.impl.HintRenderer;
+import com.intellij.diff.DiffVcsDataKeys;
+import com.intellij.diff.FrameDiffTool;
+import com.intellij.diff.contents.DiffContent;
+import com.intellij.diff.requests.DiffRequest;
+import com.intellij.diff.requests.SimpleDiffRequest;
+import com.intellij.diff.tools.util.side.OnesideTextDiffViewer;
+import com.intellij.diff.tools.util.side.ThreesideTextDiffViewer;
+import com.intellij.diff.tools.util.side.TwosideTextDiffViewer;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.editor.FoldRegion;
+import com.intellij.openapi.editor.ex.EditorEx;
+import com.intellij.openapi.util.Pair;
+import com.intellij.openapi.vcs.FilePath;
+import com.intellij.openapi.vcs.history.VcsRevisionNumber;
+import com.intellij.psi.PsiDocumentManager;
+import com.intellij.psi.PsiFile;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.research.refactorinsight.adapters.RefactoringType;
+import org.jetbrains.research.refactorinsight.data.RefactoringInfo;
+import org.jetbrains.research.refactorinsight.folding.handlers.ExtractOperationFoldingHandler;
+import org.jetbrains.research.refactorinsight.folding.handlers.FoldingHandler;
+import org.jetbrains.research.refactorinsight.folding.handlers.InlineOperationFoldingHandler;
+import org.jetbrains.research.refactorinsight.folding.handlers.MoveOperationFoldingHandler;
+import org.jetbrains.research.refactorinsight.services.MiningService;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class RefactoringFolder {
+  static Map<RefactoringType, FoldingHandler> foldingHandlers;
+
+  static {
+    foldingHandlers = new EnumMap<>(RefactoringType.class);
+    FoldingHandler moveOperationHandler = new MoveOperationFoldingHandler();
+    foldingHandlers.put(RefactoringType.MOVE_OPERATION, moveOperationHandler);
+    foldingHandlers.put(RefactoringType.PULL_UP_OPERATION, moveOperationHandler);
+    foldingHandlers.put(RefactoringType.PUSH_DOWN_OPERATION, moveOperationHandler);
+    FoldingHandler inlineOperationHandler = new InlineOperationFoldingHandler();
+    foldingHandlers.put(RefactoringType.INLINE_OPERATION, inlineOperationHandler);
+    foldingHandlers.put(RefactoringType.MOVE_AND_INLINE_OPERATION, inlineOperationHandler);
```

Comment:
Do we actually process this one?

Reply:
In new version, yes. In previous it does not work